modules/services/gitlab-runner.nix

```diff
@@ -0,0 +1,643 @@
+            '';
+          };
+          requestConcurrency = mkOption {
+            type = types.int;
+            default = 0;
+            description = ''
+              Limit number of concurrent requests for new jobs from GitLab.
+            '';
+          };
+          maximumTimeout = mkOption {
+            type = types.int;
+            default = 0;
+            description = ''
+              What is the maximum timeout (in seconds) that will be set for
+              job when using this Runner. 0 (default) simply means don't limit.
+            '';
+          };
+          protected = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              When set to true Runner will only run on pipelines
+              triggered on protected branches.
+            '';
+          };
+          debugTraceDisabled = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              When set to true Runner will disable the possibility of
+              using the <literal>CI_DEBUG_TRACE</literal> feature.
+            '';
+          };
+        };
+      });
+    };
+  };
+  config = mkIf cfg.enable {
+
+    users.users.gitlab-runner =
+      { name = "gitlab-runner";
+        uid = mkDefault 532;
+        gid = mkDefault config.users.groups.gitlab-runner.gid;
+        home = mkDefault "/var/lib/gitlab-runner";
+        shell = "/bin/bash";
+        description = "Gitlab agent user";
+      };
+    users.groups.gitlab-runner =
+      { name = "gitlab-runner";
+        gid = mkDefault 532;
+        description = "Gitlab agent user group";
+      };
+
+
+    system.activationScripts.preActivation.text = let user = config.users.users.gitlab-runner; in ''
+      mkdir -p '${user.home}'
+      chown ${toString user.uid}:${toString user.gid} '${user.home}'
+    '';
+
+
+    warnings = optional (cfg.configFile != null) "services.gitlab-runner.`configFile` is deprecated, please use services.gitlab-runner.`services`.";
+    environment.systemPackages = [ cfg.package ];
+
+    launchd.daemons.gitlab-runner = {
+      environment = { #config.networking.proxy.envVars // {
+        HOME = "${config.users.users.gitlab-runner.home}";
+        NIX_SSL_CERT_FILE = "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt";
+      } // (if config.nix.useDaemon then { NIX_REMOTE = "daemon"; } else {});
+      path = with pkgs; [
+        bash
+        gawk
+        jq
+        moreutils
+        remarshal
+        util-linux
+        cfg.package
+        coreutils
+        gnugrep
+        gnused
+      ] ++ cfg.extraPackages;
+
+        script = ''
+            ${configureScript}/bin/gitlab-runner-configure && ${startScript}/bin/gitlab-runner-start
+        '';
+
+        serviceConfig = {
+            ProcessType = "Interactive";
+            ThrottleInterval = 30;
+
+          StandardOutPath = "/var/lib/gitlab-runner/out.log";
```

Comment:
Note that launchd doesn't do any logfile rotation, unless there's a very specific reason it's probably better to use syslog by default.